src/test/java/uk/gov/companieshouse/web/accounts/interceptor/CompanyAccountsDataStateInterceptorTests.java

```diff
@@ -0,0 +1,392 @@
+package uk.gov.companieshouse.web.accounts.interceptor;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.security.SignatureException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.ModelAndView;
+import uk.gov.companieshouse.web.accounts.model.state.CompanyAccountsDataState;
+import uk.gov.companieshouse.web.accounts.model.state.CompanyAccountsDataStates;
+import uk.gov.companieshouse.web.accounts.token.TokenManager;
+
+@ExtendWith(MockitoExtension.class)
+public class CompanyAccountsDataStateInterceptorTests {
+
+    private static final String STATE_COOKIE_NAME = "__CAS";
+
+    private static final String STATE_COOKIE_VALUE = "stateCookieValue";
+
+    private static final String STATE_REQUEST_ATTRIBUTE = "companyAccountsDataState";
+
+    private static final String COMPANY_NUMBER = "companyNumber";
+
+    private static final String TRANSACTION_ID = "transactionId";
+
+    private static final String COMPANY_ACCOUNTS_ID = "companyAccountsId";
+
+    private static final String CRITERIA_URI = "/company/" + COMPANY_NUMBER + "/small-full/criteria";
+
+    private static final String BALANCE_SHEET_URI = "/company/" + COMPANY_NUMBER +
+                                                    "/transaction/" + TRANSACTION_ID +
+                                                    "/company-accounts/" + COMPANY_ACCOUNTS_ID +
+                                                    "/small-full/criteria";
+
+    private static final String APPROVAL_URI = "/company/" + COMPANY_NUMBER +
+                                                "/transaction/" + TRANSACTION_ID +
+                                                "/company-accounts/" + COMPANY_ACCOUNTS_ID +
+                                                "/small-full/approval";
+
+    @Mock
+    private HttpServletRequest httpServletRequest;
```

Comment:
For additional clarity it would be good if Mock objects were prefixed/suffixed with 'mock' throughout the file.

Reply:
Done 👍 I've opted for the prefix - I feel like `when(mockRequest...` reads best.